src/backend/access/external/libchurl.c

```diff
@@ -815,42 +815,49 @@ void check_response_code(churl_context* context)
 	}
 	else if (response_code != 200 && response_code != 100)
 	{
-		if (!handle_special_error(response_code))
-		{
-			StringInfoData err;
-			char    *http_error_msg;
-			char    *addr;
+		StringInfoData err;
+		char    *http_error_msg;
+		char    *addr;
 
-			initStringInfo(&err);
+		initStringInfo(&err);
 
-			/* prepare response text if any */
-			if (context->download_buffer->ptr)
-			{
-				context->download_buffer->ptr[context->download_buffer->top] = '\0';
-				response_text = context->download_buffer->ptr + context->download_buffer->bot;
-			}
+		/* prepare response text if any */
+		if (context->download_buffer->ptr)
+		{
+			context->download_buffer->ptr[context->download_buffer->top] = '\0';
+			response_text = context->download_buffer->ptr + context->download_buffer->bot;
+		}
 
-			/* add remote http error code */
-			appendStringInfo(&err, "remote component error (%ld)", response_code);
+		/* add remote http error code */
+		appendStringInfo(&err, "remote component error (%ld)", response_code);
 
-			addr = get_dest_address(context->curl_handle);
-			if (strlen(addr) != 0)
-				appendStringInfo(&err, " from %s", addr);
-			pfree(addr);
+		addr = get_dest_address(context->curl_handle);
+		if (strlen(addr) != 0)
+		{
+			appendStringInfo(&err, " from %s", addr);
```

Comment:
hawq codebase seems to follow convention of not using braces on single statement action on conditional blocks
if(case)
    code